File: src/main/java/com/discordclone/model/Message.java
package com.discordclone.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "messages")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Message {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false, columnDefinition = "TEXT")
    private String content;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "channel_id", nullable = false)
    private Channel channel;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User sender;
    
    @Column(nullable = false)
    private LocalDateTime timestamp = LocalDateTime.now();
    
    private boolean edited = false;
} 

File: frontend/src/types/message.ts
import { User } from './auth';

export interface Message {
    id: string;
    content: string;
    timestamp: string;
    author: User;
    channelId: string;
    attachments?: string[];
}

export interface MessageRequest {
    dm: boolean;
    content: string;
    channelId: string;
    receiver?:string;
    attachments?: string[];
}

export interface MessageResponse extends Message {
    edited: boolean;
    editedAt?: string;
}
